plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/TrinoFileSystemCache.java

```diff
@@ -64,6 +64,8 @@
 
     private final AtomicLong unique = new AtomicLong();
 
+    private static final TrinoFileSystemCacheStats STATS = new TrinoFileSystemCacheStats();
```

Comment:
Adding a unit test which ensures that the cache size from `TrinoFileSystemCacheStats` is in sync with the `map` from `TrinoFileSystemCache` would be welcome.

Reply:
Good idea. Note that it's easy to make such a test be flaky, because `TrinoFileSystemCache` is a shared mutable global state. Unless for test purposes we use a new instance of `TrinoFileSystemCacheStats` (not `TrinoFileSystemCacheStats.INSTANCE`). This requires STATS to be non-static, which we should do nonetheless.

Reply:
@findinpath @findepi I added a unit test. Please let me know if this test looks OK to you as I wasn't too sure if the approach I took is best or not.